### PR TITLE
Also forward -connect, fix #607

### LINKED ISSUE #607: Command forwarding does not forward `-connect`
When command forwarding works (see #78), `-connect` isn't properly forwarded. Commands like `+connect` or `+devmap plat23` are forwarded, not `-connect`. It means one cannot join a server by clicking on an `unv://` link while the game is already started.

### PR DESCRIPTION
**Pull request by illwieckz**
Also forward `-connect`, fix:

- #607

### CODE REVIEW

**Comment by illwieckz** [comment]
Also the transformation of `-connect` to `+connect` is now implemented in one file instead of five.

**Comment by illwieckz** [comment]
Also added fix and improvement to command-line help.

File: src/engine/client/ClientApplication.cpp [hunk 1]
```diff
@@ -48,9 +48,11 @@ class ClientApplication : public Application {
             #endif
             #ifdef BUILD_GRAPHICAL_CLIENT
                 traits.isClient = true;
+                traits.supportsUri = true;
```
Comment (slipher) [inline src/engine/client/ClientApplication.cpp:51]: Since it's true for all clients, it doesn't need to go in an ifdef.

Comment (illwieckz) [reply src/engine/client/ClientApplication.cpp]: Moved out of ifdef.

**Comment by illwieckz** [comment]
Any other comment?

File: src/engine/framework/System.cpp [hunk 2]
```diff
@@ -470,22 +470,33 @@ static void ParseCmdline(int argc, char** argv, cmdlineArgs_t& cmdlineArgs)
...
+				"  -resetconfig             reset all cvars and keybindings to their default value\n"
 #ifdef USE_CURSES
-			       "  -curses                  activate the curses interface\n"
+				"  -curses                  activate the curses interface\n"
 #endif
-			       "  -set <variable> <value>  set the value of a cvar\n"
-			       "  +<command> <args>        execute an ingame command after startup\n"
+				"  -set <variable> <value>  set the value of a cvar\n");
+			printf("%s", Application::GetTraits().supportsUri ?
+				"  -connect <" URI_SCHEME "address[:port]>\n"
```
Comment (slipher) [inline src/engine/framework/System.cpp:488]: This shouldn't have angle brackets around the whole thing. Maybe around `address` and `port`.

File: src/engine/framework/System.cpp [hunk 3]
```diff
@@ -470,22 +470,33 @@ static void ParseCmdline(int argc, char** argv, cmdlineArgs_t& cmdlineArgs)
...
-		if (!strcmp(argv[i], "--help") || !strcmp(argv[i], "-help")) {
-			std::string helpUrl = Application::GetTraits().supportsUri ? " | -connect " URI_SCHEME "ADDRESS[:PORT]" : "";
-			printf("Usage: %s [-OPTION]... [+COMMAND...%s]\n", argv[0], helpUrl.c_str());
-			printf("Possible options are:\n"
-			       "  -homepath <path>         set the path used for user-specific configuration files and downloaded dpk files\n"
-			       "  -libpath <path>          set the path containing additional executables and libraries\n"
-			       "  -pakpath <path>          add another path from which dpk files are loaded\n"
-			       "  -resetconfig             reset all cvars and keybindings to their default value\n"
+		if (!strcmp(argv[i], "--help") || !strcmp(argv[i], "-help") || !strcmp(argv[i], "-h")) {
+			printf("Usage: %s [-OPTION] [+COMMAND...]\n", argv[0]);
```
Comment (slipher) [inline src/engine/framework/System.cpp:474]: lost the `...` on options

**Comment by illwieckz** [comment]
Addressed the various comments.

File: src/engine/framework/System.cpp [hunk 4]
```diff
@@ -470,22 +470,32 @@ static void ParseCmdline(int argc, char** argv, cmdlineArgs_t& cmdlineArgs)
...
+				"  -resetconfig             reset all cvars and keybindings to their default value\n"
 #ifdef USE_CURSES
-			       "  -curses                  activate the curses interface\n"
+				"  -curses                  activate the curses interface\n"
 #endif
-			       "  -set <variable> <value>  set the value of a cvar\n"
-			       "  +<command> <args>        execute an ingame command after startup\n"
+				"  -set <variable> <value>  set the value of a cvar\n");
+			printf("%s", Application::GetTraits().supportsUri ?
+				"  -connect " URI_SCHEME "<address[:port]>\n"
```
Comment (slipher) [inline src/engine/framework/System.cpp:488]: Sorry I meant around `<address>` and `<port>` each individually.

File: src/engine/framework/System.cpp [hunk 5]
```diff
@@ -470,22 +470,32 @@ static void ParseCmdline(int argc, char** argv, cmdlineArgs_t& cmdlineArgs)
...
-		if (!strcmp(argv[i], "--help") || !strcmp(argv[i], "-help")) {
-			std::string helpUrl = Application::GetTraits().supportsUri ? " | -connect " URI_SCHEME "ADDRESS[:PORT]" : "";
-			printf("Usage: %s [-OPTION]... [+COMMAND...%s]\n", argv[0], helpUrl.c_str());
-			printf("Possible options are:\n"
-			       "  -homepath <path>         set the path used for user-specific configuration files and downloaded dpk files\n"
-			       "  -libpath <path>          set the path containing additional executables and libraries\n"
-			       "  -pakpath <path>          add another path from which dpk files are loaded\n"
-			       "  -resetconfig             reset all cvars and keybindings to their default value\n"
+		if (!strcmp(argv[i], "--help") || !strcmp(argv[i], "-help") || !strcmp(argv[i], "-h")) {
+			printf("Usage: %s [-OPTION]... [%s+COMMAND...]\n", argv[0], Application::GetTraits().supportsUri ? "-connect... | " : "");
```
Comment (slipher) [inline src/engine/framework/System.cpp:474]: `-connect <uri>` instead of `-connect...`?

File: src/engine/framework/System.cpp (hunk 4 above)
Comment (illwieckz) [reply src/engine/framework/System.cpp]: This way?

Comment (slipher) [reply src/engine/framework/System.cpp]: Yes, LGTM

**Pull request merged**